internal/nix/nixprofile/profile.go

```diff
@@ -26,18 +28,62 @@ func ProfileListItems(
 	profileDir string,
 ) (map[string]*NixProfileListItem, error) {
 
-	lines, err := nix.ProfileList(writer, profileDir)
+	version, err := nix.Version()
 	if err != nil {
 		return nil, err
 	}
 
+	// if version is >= 2.17, we can use the json output
+	useJSON := vercheck.SemverCompare(version, "2.17") >= 0
```

Comment:
Use `2.17.0` (2.17 is technically not valid semver) 

This defaults to false if version is not semver. Is that OK?  (I think probably yes)

Reply:
ugh... good point

let me check the behavior of this `vercheck.SemverCompare` for `2.17` like output.

btw, do we know if `nix --version` prints `2.17` or `2.17.0`?

Reply:
2.17.0

<img width="333" alt="image" src="https://github.com/jetpack-io/devbox/assets/544948/48ac09ba-3891-4ff7-aa46-c27eadcb269d">

`vercheck.SemverCompare` seems to work because `semver.parse` can read versions missing the patch and defaults them to `0`. But I'm not sure that a missing patch number is generally considered semver.